.circleci/config.yml

```diff
@@ -28,6 +28,18 @@ commands:
             ACTUAL_SIGNATURE=$(php -r "echo hash_file('sha384', 'composer-setup.php');")
             [[ "$EXPECTED_SIGNATURE" == "$ACTUAL_SIGNATURE" ]] && sudo php composer-setup.php --install-dir=/bin --filename=composer || exit 1
 
+  conditionally-skip:
+    description: "Skip if the current commit has no change in relevant directories"
+    parameters:
+      directories:
+        type: string
+    steps:
+      - run:
+          name: Checking whether relevant files were changed
+          command: |
+            CHANGED_FILES="$( git diff HEAD^ HEAD --name-only --diff-filter=ACMRT << parameters.directories >> )"
```

Comment:
This exits early from the job if there's no relevant file in the latest commit.

Thanks to Nick in this [support ticket](https://support.circleci.com/hc/en-us/requests/79646?page=1).

It may be a little confusing, as the green check will still show for the job, even though it was skipped.

But this is probably a big improvement in built times. This can skip 6 PHPUnit test jobs for JS-only commits.

Reply:
Good call out — I agree that the trade-off of having a green tick even if the job didn't run is worth it for the build time improvement.